src/Generator/Generators/CLI/CLIHeadersTemplate.cs

```diff
@@ -137,6 +137,9 @@ public void GenerateForwardRefs()
             var typeReferences = typeReferenceCollector.TypeReferences;
             var @namespace = ConvertForwardReferencesToNamespaces(typeReferences);
 
+            foreach (var cls in @namespace.Classes)
+                cls.IsStruct = ExtensionMethods.CheckClassIsStructible(cls, Driver);
```

Comment:
Thanks. Will try to implement it.
On 01-Jul-2015 5:55 PM, "Dimitar Dobrev" <notifications@github.com> wrote:

> In src/Generator/Generators/CLI/CLIHeadersTemplate.cs
> <https://github.com/mono/CppSharp/pull/505#discussion_r33672942>:
>
> > @@ -137,6 +137,9 @@ public void GenerateForwardRefs()
> >              var typeReferences = typeReferenceCollector.TypeReferences;
> >              var @namespace = ConvertForwardReferencesToNamespaces(typeReferences);
> >
> > +            foreach (var cls in @namespace.Classes)
> > +                cls.IsStruct = ExtensionMethods.CheckClassIsStructible(cls, Driver);
>
> You just add the pass to the list of passes in Driver.SetupPasses.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mono/CppSharp/pull/505/files#r33672942>.
>